docs/style-guide.rst

```diff
@@ -701,7 +701,65 @@ indistinguishable from the numerals one and zero.
 Contract and Library Names
 ==========================
 
-Contracts and libraries should be named using the CapWords style. Examples: ``SimpleToken``, ``SmartBank``, ``CertificateHashRepository``, ``Player``.
+Contracts and libraries should be named using the CapWords style. Examples: ``SimpleToken``, ``SmartBank``, ``CertificateHashRepository``, ``Player``. Contract and library names should match their Contructor name and their filenames.
```

Comment:
Can you clarify this a bit? I think it should read: `Contract and library names must match their Constructor name and should match their filenames.`

Reply:
Not sure why we would capitalize "Constructor" here...

Reply:
@axic In the demo that I've pushed here https://github.com/ltfschoen/dao, if the Contract Name and Library Name are both CapWords `contract Owned {` in Owned.sol and `contract Congress is Owned, TokenRecipient {` in Congress.sol, and the Contract Filename and Library Name are both CapWords as Owned.sol and Congress.sol, but I change the Contract Constructor Name and the Library Constructor Name to Lowercase so they become `function owned()  public {` in Owned.sol and `function congress (` in Congress.sol, then it still compiles and deploys successfully. That's why I've written "Contract and library names **should** match their Constructor" (since it's ideal if they do match, but it still appears to compile if they don't).

If I restore them back to CapWords, and then rename the Contract Filename and Library Filename from Congress.sol to congress.sol, and Owned.sol to owned.sol, and update the contract so it imports the library as lowercase, and update the migration file so it imports the lowercase name, then it still compiles and deploys successfully, so I think it shold say "Contract and library names **should** match their filenames".

I'm going to change it to `Contract and library names should match their Constructor name and should also match their filenames.` (i.e. if the Contract name is `Congress` and the library name is `Owned`, then their associated Constructor names should be written in CapWords as `Congress` and `Owned` respectively, and their associated filenames should be Congress.sol and Owned.sol), what do you think?

Reply:
If you specify the relation between constructor names and contract name you cannot use the word "should", it isn't an optional relation, the word "must" must (:wink:) be used.